pkg/server/debug/pprofui/server.go

```diff
@@ -244,6 +300,18 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// If the profile is not in a protobuf format, downloading it is the only
+	// option.
+	if !isProfileProto {
```

Comment:
I've seen this before, why do we have this twice?

Reply:
This logic exists to serve a newly generated profile. The same logic also exists to serve a cached profile request by profile ID.